kotlinx-coroutines-core/jvm/src/scheduling/WorkQueue.kt

```diff
@@ -16,6 +16,11 @@ internal const val MASK = BUFFER_CAPACITY - 1 // 128 by default
 internal const val TASK_STOLEN = -1L
 internal const val NOTHING_TO_STEAL = -2L
 
+internal typealias StealingMode = Int
+internal const val STEAL_ANY: StealingMode = -1
+internal const val STEAL_CPU_ONLY: StealingMode = 0
+internal const val STEAL_BLOCKING_ONLY: StealingMode = 1
```

Comment:
Potential simplification, up to you whether to use it:

```suggestion
internal const val STEAL_ANY: StealingMode = 3
internal const val STEAL_CPU_ONLY: StealingMode = 2
internal const val STEAL_BLOCKING_ONLY: StealingMode = 1
```
This could simplify the code a bit with bitmasks.

```kotlin
// maybe inline
val Task.mode: Int
  get() = if (isBlocking) STEAL_BLOCKING_ONLY else STEAL_CPU_ONLY
```
Then,
```kotlin
            if (lastScheduled.isBlocking) {
                if (stealingMode == STEAL_CPU_ONLY) {
                    return NOTHING_TO_STEAL
                }
            } else if (stealingMode == STEAL_BLOCKING_ONLY) {
                return NOTHING_TO_STEAL
            }
```
becomes
```kotlin
if ((lastScheduled.mode and stealingMode) == 0)
  return NOTHING_TO_STEAL
```